In [6]:
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf

In [3]:
df = pd.read_csv('../student-mat.csv', delimiter=';')

In [4]:
row_names = df.index.tolist()
column_names = df.columns.tolist()

In [5]:
age_15_studytime_data = []
age_15_g1_data = []
age_15_g2_data = []
age_15_g3_data = []

for row in row_names:
    if df.loc[row, 'age'] == 15:
        studytime = df.loc[row, 'studytime']
        g1 = df.loc[row, 'G1']
        g2 = df.loc[row, 'G2']
        g3 = df.loc[row, 'G3']
        age_15_g1_data.append(g1)
        age_15_g2_data.append(g2)
        age_15_g3_data.append(g3)
        age_15_studytime_data.append(studytime)

In [7]:
model = smf.ols('G3 ~ studytime', data=df).fit()

In [8]:
anova_table = sm.stats.anova_lm(model, typ=2)
print(anova_table)

print(f"Regression Equation: G3 = {model.params['Intercept']} + {model.params['studytime']}*studytime")

                sum_sq     df         F    PR(>F)
studytime    79.132208    1.0  3.796826  0.052061
Residual   8190.776653  393.0       NaN       NaN
Regression Equation: G3 = 9.328260988559865 + 0.5340011312423562*studytime


In [9]:
print(model.summary())

SST = model.centered_tss
SSR = model.ess
SSE = model.ssr
R_squared = model.rsquared

print(f"SST (Total Sum of Squares): {SST}")
print(f"SSR (Regression Sum of Squares): {SSR}")
print(f"SSE (Error Sum of Squares): {SSE}")
print(f"R-squared (Coefficient of Determination): {R_squared}")

                            OLS Regression Results                            
Dep. Variable:                     G3   R-squared:                       0.010
Model:                            OLS   Adj. R-squared:                  0.007
Method:                 Least Squares   F-statistic:                     3.797
Date:                Fri, 12 Apr 2024   Prob (F-statistic):             0.0521
Time:                        18:17:12   Log-Likelihood:                -1159.3
No. Observations:                 395   AIC:                             2323.
Df Residuals:                     393   BIC:                             2331.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      9.3283      0.603     15.463      0.0

The goodness of the coefficient of determination is low as the number is 0.010. This means that the dependent variable is not in proportion to the independent variable. However, this makes sense due to the amount of independent variables.